src/Microsoft.TemplateEngine.Core/VariableCollection.cs

```diff
@@ -16,7 +16,7 @@ namespace Microsoft.TemplateEngine.Core
     public class VariableCollection : IVariableCollection
```

Comment:
If we want to support explicit null values - then `VariableCollectionFromParameters` needs to be altered as well (not sure if there are other places filtering nulls).

Though I'm not 100% sure we need explicit null values - as it should behave exactly as missing value. So it seems to only add complexity but not an actual value.

Reply:
I'm not strong about it. I fixed it other way. I think it's better to avoid `null`